lib/cext/preprocess.rb

```diff
@@ -30,6 +35,36 @@
 
       "#{shim}(#{argc}, #{argv}, \"#{arity}\", "
     }
+  elsif line =~ /^(\s+)VALUE((\s+\w+\s*(\[\s*\d+\s*\]\s*)?,?)+);\s*$/
+    # Translate
+    #   VALUE args[6], failed, a1, a2, a3, a4, a5, a6;
+    #  into
+    #   VALUE failed, a1, a2, a3, a4, a5, a6; VALUE *args = truffle_managed_malloc(6 * sizeof(VALUE));
+
+    simple = []
+    arrays = []
+
+    line = $1
+
+    $2.split(', ').each do |local|
+      local.strip!
+      if local.end_with?(']')
+        local =~ /(\w+)\s*\[\s*(\d+)\s*\]/
+        arrays.push [$1, $2.to_i]
```

Comment:
I would personally consider `push` only for `Queue#push`. I would use `Array#concat(array)` rather than `Array#push(*array)` if the operand is an Array. There are few cases to append two elements at once and even then `<<` can be chained: `ary << 1 << 2`.
I find `push` confusing for this usage, yes. It might be more meaningful when using an Array as a stack with `push`/`pop` but here it's really "append to a list of things", that is `<<`.

But more importantly it seems extremely clear from the statistics above the idiomatic way is to use the operator here (I think that's a general rule, except for `Array#*(String)`).
I think it's good we agree on the same way to do things within our own core and related Ruby code.